Enable discoverable devices probe kselftest on MediaTek chromebooks

nfraprado opened the pull request:
Enable the discoverable devices probe kselftest on the MediaTek Chromebook platforms. This test will validate that the devices connected through USB or PCI on those platforms have been probed successfully. The devices are, namely, the camera on mt8183-juniper and mt8186-steelix, and the camera, wifi and bluetooth on mt8195-tomato.

Depends on https://github.com/kernelci/kernelci-core/pull/2577.

Fixes #579 .

nfraprado commented:
Manual LAVA runs for reference:
mt8195-tomato: https://lava.collabora.dev/scheduler/job/14310494
mt8183-juniper: https://lava.collabora.dev/scheduler/job/14310955
mt8186-steelix: https://lava.collabora.dev/scheduler/job/14310717

pawiecz commented on config/pipeline.yaml:
```diff
@@ -447,6 +447,20 @@ jobs:
     kcidb_test_suite: kselftest.dt
 
+  kselftest-discoverable-devices-probe:
+    template: kselftest.jinja2
```
Depending on when this PR gets merged, be advised that the job template might need some adjustment; see https://github.com/kernelci/kernelci-pipeline/pull/655

pawiecz replied:
This job template no longer exists. It was replaced by [`generic.jinja2`](https://github.com/kernelci/kernelci-pipeline/blob/main/config/runtime/generic.jinja2) - see [the example](https://github.com/kernelci/kernelci-pipeline/blob/main/doc/developer-documentation.md#an-example-of-enabling-a-new-job) for more details.

nfraprado replied:
Updated.

pawiecz commented on config/pipeline.yaml:
```diff
@@ -447,6 +447,20 @@ jobs:
 
+  kselftest-discoverable-devices-probe:
+    template: kselftest.jinja2
+    kind: test
```
Is this a single test case or a test suite? It [might need adjusting its `kind`](https://github.com/kernelci/kernelci-pipeline/pull/648#discussion_r1665524836); see https://github.com/kernelci/kernelci-pipeline/commit/396c67d0ba5260a56b4c4ceccd1635665150f527

JenySadadia replied:
I agree with @pawiecz's suggestion. It should have `kind: job` in case of a test suite.

nfraprado replied:
It is a test suite. Fixed.

pawiecz commented on config/pipeline.yaml:
```diff
@@ -447,6 +447,20 @@ jobs:
+    template: kselftest.jinja2
+    kind: test
+    params:
+      nfsroot: 'http://storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/20240221.0/{debarch}' # TODO: update once new rootfs with platform-test-parameters is generated
```
Try https://storage.staging.kernelci.org/rootfs/bookworm-kselftest/20240704.0 (just to spin this up on staging)

nfraprado replied:
Done. Leaving the comment so I remember to update it to a production URL before merging.

pawiecz commented on config/scheduler-chromeos.yaml:
```diff
@@ -187,6 +187,14 @@ scheduler:
+    <<: *lava-job-collabora
+    event:
+      channel: node
+      name: kbuild-gcc-10-arm64-chromebook
```
GCC was recently bumped to 12; see https://github.com/kernelci/kernelci-pipeline/commit/ac15e371e2d5cf8bd1bdc621493892346658f0e6

nfraprado replied:
Fixed.

pawiecz reviewed (commented):
LGTM (left a few minor comments)

musamaanjum commented on config/scheduler-chromeos.yaml:
```diff
@@ -187,6 +187,14 @@ scheduler:
       result: pass
     platforms: *mediatek-platforms
 
+  - job: kselftest-discoverable-devices-probe
```
Is `discoverable-devices-probe` a kselftest? If not, why the same naming layout is being used?

nfraprado replied:
It is a kselftest: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=dacf1d7a78bf8a131346c47bfba7fe1f3ff44beb

nfraprado commented:
Since the dependencies are available in staging I've removed staging-skip label.

nfraprado commented:
I just noticed that I forgot to document that this change also depends on a kernel series: https://lore.kernel.org/all/20240613-kselftest-discoverable-probe-mt8195-kci-v1-0-7b396a9b032d@collabora.com

It is present on linux-next, so I've updated the `rules` for the job accordingly so the PR can already be tested and will run only on next. Also added a comment about how to change the rule in ~3 weeks from now to start running this test on mainline as well.

musamaanjum commented on config/pipeline.yaml:
```diff
@@ -857,6 +857,21 @@ jobs:
         patchlevel: 7
     kcidb_test_suite: kselftest.dt
 
+  kselftest-discoverable-devices-probe:
```
We usually use only the name of the test suite at the end of the job name. Maybe better to follow the same naming scheme and rename it to just `kselftest-discoverable-devices` as suite name is `devices`. The same rename should be done for kcidb_test_suite property.

nfraprado replied:
Updated the name to match the kselftest collection like the others.

nfraprado commented:
I've updated the kselftest collection name to `devices/probe` following the move in https://lore.kernel.org/all/20240705-dev-err-log-selftest-v2-1-163b9cd7b3c1@collabora.com

nfraprado commented:
I've generated the rootfs on staging for https://github.com/kernelci/kernelci-core/pull/2577 and updated the URL here, so removing the staging-skip tag so this can get tested.

nfraprado commented:
Tested on all platforms.

MT8186 OK:
https://lava.collabora.dev/scheduler/job/15670426

MT8183 OK:
https://lava.collabora.dev/scheduler/job/15672968

MT8192 not OK:
https://lava.collabora.dev/scheduler/job/15672970
The board file was not present in repo. Now merged through https://github.com/kernelci/platform-test-parameters/pull/2.

MT8195 partial OK:
https://lava.collabora.dev/scheduler/job/15672971
USB devices were failing [due to unmerged PR](https://github.com/kernelci/platform-test-parameters/pull/1), has been merged now.

So once https://github.com/kernelci/kernelci-core/pull/2577 is merged and the rootfs available in the production storage, its URL can be updated here and this PR merged.

nuclearcat commented:
Can you please rebase and let's merge?

nfraprado commented:
@nuclearcat I also need to update the rootfs URL to the production rootfs URL for bookworm-kselftest, but the latest in https://storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/ is still `20240313.0/`, so it seems that the rootfs hasn't been generated yet with https://github.com/kernelci/kernelci-core/pull/2577 merged. Does the production rootfs generation need to be done manually?

musamaanjum commented:
@nfraprado we have CI for this here (https://github.com/kernelci/kernelci-core/actions/workflows/rootfs.yml). Not sure if you require permissions to run it. 

Click on "Run Workflow". Select the branch, rootfs name, architecture and destination. The destination must be staging for verification of rootfs. Once you verify the results of your test on staging, create the rootfs on production and then merge the PR.

nfraprado commented:
Hm, I just ran it here: https://github.com/kernelci/kernelci-core/actions/runs/11165276908/job/31036570425

But still don't see it in https://storage.staging.kernelci.org/images/rootfs/debian/bookworm-kselftest/

musamaanjum commented:
![Screenshot from 2022-10-05 13-10-23](https://github.com/user-attachments/assets/124dbeb7-bb4e-4cc4-a24c-6e515fe57b95)
I've created a job (https://github.com/kernelci/kernelci-core/actions/runs/11176504665) as well with the parameters mentioned in the above image. I'm unable to find the artifacts on storage.staging.

I'll leave it and let it get debugged by original authors.

nuclearcat commented:
https://storage.staging.kernelci.org/rootfs/bookworm-kselftest/

nfraprado commented:
Tested OK: https://lava.collabora.dev/scheduler/job/15963263

So I've deployed the rootfs to production: https://github.com/kernelci/kernelci-core/actions/runs/11220957370

And updated the PR here to use the proper URL for the production rootfs. So now it's ready to merge.
